test/byContract/IexecPoco/IexecPoco2-contribute-and-finalize.test.ts

```diff
@@ -0,0 +1,515 @@
+                        await buildAndSignContributionAuthorizationMessage(
+                            worker.address,
+                            taskId,
+                            emptyEnclaveAddress,
+                            scheduler,
+                        ),
+                    ),
+            ).to.be.revertedWithoutReason(); // require#5
+        });
+
+        it('Should not contributeAndFinalize when enclave challenge is missing (TEE)', async () => {
+            const { dealId, taskIndex, taskId } = await iexecWrapper.signAndMatchOrders(
+                ...buildOrders({
+                    assets: ordersAssets,
+                    requester: requester.address,
+                    prices: ordersPrices,
+                    volume,
+                    trust,
+                    tag: teeDealTag,
+                }).toArray(),
+            );
+            await iexecPoco.initialize(dealId, taskIndex).then((tx) => tx.wait());
+            // Task active, before deadline, good trust, but missing enclave challenge.
+            await expect(
+                iexecPoco
+                    .connect(worker)
+                    .contributeAndFinalize(
+                        taskId,
+                        resultDigest,
+                        results,
+                        noCallbackData,
+                        emptyEnclaveAddress,
+                        emptyEnclaveSignature,
+                        await buildAndSignContributionAuthorizationMessage(
+                            worker.address,
+                            taskId,
+                            emptyEnclaveAddress,
+                            scheduler,
+                        ),
+                    ),
+            ).to.be.revertedWithoutReason(); // require#6
+        });
+
+        it('Should not contributeAndFinalize when scheduler signature is invalid', async () => {
+            const { dealId, taskIndex, taskId } = await iexecWrapper.signAndMatchOrders(
+                ...buildOrders({
+                    assets: ordersAssets,
+                    requester: requester.address,
+                    prices: ordersPrices,
+                    volume,
+                    trust,
+                    tag: standardDealTag,
+                }).toArray(),
+            );
+            await iexecPoco.initialize(dealId, taskIndex).then((tx) => tx.wait());
+            // Task active, before deadline, good trust, but invalid scheduler signature.
+            await expect(
+                iexecPoco.connect(worker).contributeAndFinalize(
+                    taskId,
+                    resultDigest,
+                    results,
+                    noCallbackData,
+                    emptyEnclaveAddress,
+                    emptyEnclaveSignature,
+                    '0xbadd', // Bad scheduler signature
+                ),
+            ).to.be.revertedWith('invalid-signature-format'); // require#7
+        });
+
```

Comment:
```suggestion

    it('Should not contributeAndFinalize when authorization is not signed by broker', async () => {
    
```

Reply:
[Add test for TEE broker signature](https://github.com/iExecBlockchainComputing/PoCo/pull/176/commits/12f88d357edead01a27e62e455391389171e2e4f)